### ISSUE TITLE
Non rectangular images - Coordinates - Multiple channel extraction

### ISSUE THREAD

**Issue by MathieuBo** [lens-biophotonics/ZetaStitcher#13]
Hi all, 

Thanks a lot for developping such a great tool. I have three questions:
- Can the package handle non rectangular pictures?
- What is the benefit of using stage coordinates vs tile order (1_1, 1_2, 2_1, 2_2 for example)? Will the stitching be more accurate with stage coordinates? I tried with just a rough tile ordering and it worked extremely well already.
- A channel argument can be passed to extract R,G,B channels selectively. What would be the best strategy to stitch a multichannel picture? If the yml file is generated based on a reference channel (e.g. DAPI), can ```stitch-fuse```be run on another one?

Thanks a lot for your help! 
Mathieu

**Comment by gmazzamuto** [comment]
Hi Mathieu, thanks for your interest in ZetaStitcher!

* Can you please explain better what you mean with non rectangular images?

* If you provide the `--overlap` argument (and optionally `--px-size-xy`), then as you've already found out everything should work just fine using the tile ordering in the file name.

* Yes. In `stitch-fuse`, the channel argument specifies which channel to use to compute the alignment. If no channel is specified, the channels are summed together and the resulting grayscale image is used for alignment. Once the alignment is computed, i.e. the `stitch.yml` file is generated, you can run `stitch-fuse` to create the fused image. If you specify the channel argument, only that color will be fused, otherwise all channels will be fused (this no matter what channel was used for "reference", i.e. for alignment, in `stitch-align`).

Please use the latest version 0.7.0 that I have just released. It contains some improvements in the alignment algorithm (using the difference of gaussians) and also in the RGB handling.

Let me know if you encounter any problems or if you need more help! Or if there are new issues with this latest release.. :)

Cheers,
Giacomo

**Comment by MathieuBo** [comment]
Thanks a lot @gmazzamuto! I'll definitely look into the new version. Will it be able to handle more complex images than RGB? (I am working with 7 color images) 

Regarding non-rectangular, see the example below. Right is what I meant by non-rectangular images.

Cheers, 
Mathieu

![stitching_example](https://user-images.githubusercontent.com/14175231/173024654-0ce07733-1b8e-4646-a5d6-29d5277e6812.png)

**Comment by gmazzamuto** [comment]
Yes it should work with multichannel images, just use the channel number (zero based) in the options related to channels

Unfortunately right now non-rectangular images are not supported, but it's something we urgently need also on our side so I am confident I will add this feature in the coming weeks. I'll let you know when it's out

**Comment by MathieuBo** [comment]
Hi @gmazzamuto! I was just curious to know if there were any updates on the 'non-rectangular' image stitching. Is this a feature you have implemented? 
Thanks!

**Comment by gmazzamuto** [comment]
Hi Mathieu,

sorry for the late reply but I've been busy and traveling. Unfortunately no, I haven't had the time to work on it... Would it work for you if the rectangular images were all identical to each other in size? If yes, this would make the change easier so I can try and see if I can do it

**Comment by MathieuBo** [comment]
Hi @gmazzamuto, 
No worries :)
I am not sure I follow what you mean by rectangular images identical to each other in size? 
All invididual tiles have identical sizes, I think that filling the missing ones with empty tile for a final large rectangular image would work very well. 
Thanls !!
